src/main/java/com/qunite/api/service/QueueServiceImpl.java

```diff
@@ -51,12 +53,43 @@ public Optional<Integer> getMemberPositionInQueue(Long memberId, Long queueId) {
         .map(entry -> entry.getEntryIndex() + 1);
   }
 
+  @Override
+  @Transactional(isolation = Isolation.REPEATABLE_READ)
+  public void changeMemberPositionInQueue(Long memberId, Long queueId,
+                                          Integer newIndex, String username) {
+    if (isUserQueueCreatorOrManagerByCredentials(queueId, username)) {
+      entryRepository.findById(new EntryId(memberId, queueId)).ifPresent(entry -> {
+        var currentIndex = entry.getEntryIndex();
+        if (!currentIndex.equals(newIndex)) {
+          var startIndex = Math.min(currentIndex, newIndex);
+          var endIndex = Math.max(currentIndex, newIndex);
+          var increment = currentIndex < newIndex ? -1 : 1;
+
+          entryRepository.updateEntryIndexes(queueId, startIndex, endIndex, increment);
+          entry.setEntryIndex(newIndex);
+        }
+      });
+    } else {
+      throw new AccessDeniedException("User is not a creator or manager");
+    }
+  }
+
+  @Override
+  @Transactional
+  public void deleteMemberFromQueue(Long memberId, Long queueId, String username) {
+    if (isUserQueueCreatorOrManagerByCredentials(queueId, username)) {
+      entryRepository.deleteById(new EntryId(memberId, queueId));
```

Comment:
Correct me if I am wrong: 
Lets say we have example queue:
Entry index: 0, EntryId(memberId=7, queueId=1)
Entry index: 1, EntryId(memberId=6, queueId=1)
Entry index: 2, EntryId(memberId=5, queueId=1)
Entry index: 3, EntryId(memberId=4, queueId=1)
Entry index: 4, EntryId(memberId=3, queueId=1).
We perform deleteMemberFromQueue(5, 1, "admin") and we get the following: 
Entry index: 0, EntryId(memberId=7, queueId=1)
Entry index: 1, EntryId(memberId=6, queueId=1)
Entry index: 3, EntryId(memberId=4, queueId=1)
Entry index: 4, EntryId(memberId=3, queueId=1).
So, entry index 2 is missing. And our queue now looks like this:
```
1.  member 7
2.  member 6
4.  member 4
5.  member 3
```

third place is missing. 

And my question is: Do we agree with such logic or not?